webui/src/pages/repositories/wizard/spark_quickstart_wizard.jsx

```diff
@@ -0,0 +1,87 @@
+import Wizard from "./wizard";
+import React, {useState} from "react";
+import {useAPI} from "../../../lib/hooks/api";
+import {config} from "../../../lib/api";
+import {Loading} from "../../../lib/components/controls";
+import Container from "react-bootstrap/Container";
+import Row from "react-bootstrap/Row";
+import Col from "react-bootstrap/Col";
+import {Spinner} from "react-bootstrap";
+import Alert from "react-bootstrap/Alert";
+import RepositoryCreateForm from "../../../lib/components/repositoryCreateForm";
+
+const RepositoryCreationPhase = {
+    NotStarted: 0,
+    InProgress: 1,
+    Completed: 2,
+}
+
+const RepositoryCreationStep = ({repoCreationError, createRepo, onCancel, onComplete}) => {
+    const {response, error: err, loading} = useAPI(() => config.getStorageConfig());
+    const [repoName, setRepoName] = useState('');
+    const [repoCreationPhase, setRepoCreationPhase] = useState(RepositoryCreationPhase.NotStarted);
+
+    const onSubmit = async (repo) => {
+        setRepoCreationPhase(RepositoryCreationPhase.InProgress);
+        const success = await createRepo(repo, false);
+        setRepoCreationPhase(RepositoryCreationPhase.Completed);
+        if (success) {
+            onComplete({ 'branch': repo.default_branch, 'namespace': repo.storage_namespace },);
+            setRepoName(repo.name);
+        }
+    }
+    const showError = (repoCreationError) ? repoCreationError : err;
+    let present;
+    if (loading) {
+        present = <Loading/>;
+    } else if (repoCreationPhase === RepositoryCreationPhase.InProgress) {
+        present = <Container>
+            <Row className={'justify-content-left'}>
+                <Col className={"col-1 mb-2 mt-2"}>
+                    <Spinner animation="border" variant="primary" size="xl" role="status" />
+                </Col>
+                <Col className={"col-3 mb-2 mt-2"}>
+                    <span>Creating repository...</span>
+                </Col>
+            </Row>
+        </Container>;
+    } else if (repoCreationPhase === RepositoryCreationPhase.Completed) {
```

Comment:
Is this an intentional decision to set the phase to RepositoryCreationPhase.Completed on both success and failure? Semantically, I think it makes more sense to say that the repo creation phase is completed when it succeeded (the repo step is a non-skippable one, so to complete it the user must successfully create a commit). Programmatically, I think that it also makes more sense because now you are relying on line 33 never to apply the current condition to a non-successful repo creation and someone may miss it. 
WDYT?

Reply:
Not intentional at all.
Good catch...